File: prop-all/src/test/java/io/pleo/prop/objects/MyInterfaceProvider.java
package io.pleo.prop.objects;

import javax.inject.Inject;
import javax.inject.Named;
import javax.inject.Provider;

import io.pleo.prop.core.Prop;

public class MyInterfaceProvider implements Provider<MyInterface> {
  private final Prop<String> prop;

  public Prop<String> getProp() {
    return prop;
  }

  @Inject
  public MyInterfaceProvider(@Named("io.pleo.test.prop3") Prop<String> prop) {
    this.prop = prop;
  }

  @Override

  public MyInterface get() {
    return prop::get;
  }
}

File: prop-all/src/test/kotlin/io/pleo/prop/objects/MyInterfaceProvider.kt
package io.pleo.prop.objects

import com.google.inject.Inject
import com.google.inject.name.Named
import io.pleo.prop.core.Prop
import jakarta.inject.Provider

class MyInterfaceProvider @Inject constructor(
    @Named("io.pleo.test.prop3") val prop: Prop<String>
) : Provider<MyInterface> {
    override fun get(): MyInterface =
        object : MyInterface {
            override val propValue = prop()
        }
}
